### PR TITLE
Clarify the computation of ExtLen.

### LINKED ISSUE #14: Explain&nbsp;how to calculate the length in bytes of the `ExtLen` field in the SCION Hop-by-Hop/End-to-End Options header (extension&nbsp;header of a SCION path header).

### PR DESCRIPTION
**Pull request by jiceatscion**
Fixes #14

### CODE REVIEW

File: draft-dekater-scion-dataplane.md [hunk 1]
```diff
@@ -790,8 +790,8 @@ The SCION Hop-by-Hop Options and End-to-End Options headers have the following f
 
 
 - `NextHdr`: Unsigned 8-bit integer. Identifies the type of header immediately following the Hop-by-Hop/End-to-End Options header. Values of this field respect the Assigned SCION Protocol Numbers (see also [](#protnum)).
-- `ExtLen`: Unsigned 8-bit integer. The value of this field is computed as the length of the complete Hop-by-Hop/End-to-End Options header in multiples of 4-bytes minus 1.
-- `Options`: This is a variable-length field. The length of this field MUST be such that the complete length of the Hop-by-Hop/End-to-End Options header is an integer multiple of 4 bytes. The `Options` field contains one or more Type-Length-Value (TLV) encoded options. For details, see [](#optfld).
+- `ExtLen`: Unsigned 8-bit integer. The value of this field is computed as the length of the header in 4-octet units not including the first 4 octets. That is: `ExtLen = uint8(((options_len + 2) / 4) - 1)`, where `options_len + 2` is a multiple of 4.
```
Comment (nicorusti) [inline draft-dekater-scion-dataplane.md:793]:
```suggestion
- `ExtLen`: Unsigned 8-bit integer. Length of the  Hop-by-Hop or End-to-End Options header in 4-octet units, not including the first 4 octets. That is: `ExtLen = uint8(((options_len + 2) / 4) - 1)`, where `options_len + 2` is a multiple of 4.
```


I'd try language similar to IPv6 (RFC2460): 
```   
Hdr Ext Len    8-bit unsigned integer.  Length of the Routing
                   header in 8-octet units, not including the first
                   8 octets.
```

I don't really understand the formula, and I'd not use `options_len` in there as it seems we are introducing a new field. I'd rather use tomething lile `L` and then on the side say what `L` is (extension header size in bits)

File: draft-dekater-scion-dataplane.md [hunk 2]
```diff
@@ -790,8 +790,8 @@ The SCION Hop-by-Hop Options and End-to-End Options headers have the following f
...
+- `Options`: This is a variable-length field. The length of this field MUST be such that the complete length of the Hop-by-Hop/End-to-End Options header is an integer multiple of 4 bytes. This can be achieved by using padding-only options. The `Options` field contains one or more Type-Length-Value (TLV) encoded options. For details, see [](#optfld).
```
Comment (nicorusti) [inline draft-dekater-scion-dataplane.md:794]: What is a `padding-only` option?  In the draft we describe `Pad1` and `PadN`

**Review by nicorusti — CHANGES_REQUESTED** [review]

File: draft-dekater-scion-dataplane.md (hunk 1 above)
Comment (jiceatscion) [reply draft-dekater-scion-dataplane.md:793]: I did use the language of RFC8200 for inspiration... rewrote to be even closer. The formula had to be added in order to answer the question that the reviewer asked: "how to compute it".

File: draft-dekater-scion-dataplane.md (hunk 2 above)
Comment (jiceatscion) [reply draft-dekater-scion-dataplane.md:794]: Rewrote to remove the phrase "padding-only" options and link to table-4.

**Review by nicorusti — APPROVED** [review]

**Pull request merged**